msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplication.kt

```diff
@@ -0,0 +1,111 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.identity.client
+
+import android.content.Context
+import com.microsoft.identity.client.exception.MsalClientException
+import com.microsoft.identity.client.exception.MsalException
+import com.microsoft.identity.client.internal.CommandParametersAdapter
+import com.microsoft.identity.client.statemachine.BrowserRequiredError
+import com.microsoft.identity.client.statemachine.GeneralError
+import com.microsoft.identity.client.statemachine.InvalidAttributesError
+import com.microsoft.identity.client.statemachine.InvalidEmailError
+import com.microsoft.identity.client.statemachine.InvalidPasswordError
+import com.microsoft.identity.client.statemachine.PasswordIncorrectError
+import com.microsoft.identity.client.statemachine.UserAlreadyExistsError
+import com.microsoft.identity.client.statemachine.UserNotFoundError
+import com.microsoft.identity.common.crypto.AndroidAuthSdkStorageEncryptionManager
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager
+import com.microsoft.identity.common.internal.controllers.LocalMSALController
+import com.microsoft.identity.common.internal.controllers.NativeAuthController
+import com.microsoft.identity.common.internal.net.cache.HttpCache
+import com.microsoft.identity.common.java.authorities.Authority
+import com.microsoft.identity.common.java.cache.ICacheRecord
+import com.microsoft.identity.common.java.commands.CommandCallback
+import com.microsoft.identity.common.java.controllers.CommandDispatcher
+import com.microsoft.identity.common.java.controllers.results.ICommandResult
+import com.microsoft.identity.common.java.eststelemetry.PublicApiId
+import com.microsoft.identity.common.java.exception.BaseException
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger.LogLevel
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectory
+import com.microsoft.identity.common.java.util.ResultFuture
+import com.microsoft.identity.common.java.util.checkAndWrapCommandResultType
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class NativeAuthPublicClientApplication(
+    private val nativeAuthConfig: NativeAuthPublicClientApplicationConfiguration
+) : INativeAuthPublicClientApplication, PublicClientApplication(nativeAuthConfig) {
+
+    private lateinit var sharedPreferencesFileManager: SharedPreferencesFileManager
+
+    init {
+        initializeApplication()
+        initializeSharedPreferenceFileManager(nativeAuthConfig.appContext)
```

Comment:
I suspect you need this because you want to operate in SINGLE account mode

Reply:
@SammyO

Reply:
@shahzaibj thanks for flagging this, this is something I had wanted to discuss with you. 
So this SharedPreferencesFileManager is only used for multi-account flows? We don't need it if native auth is operating in a single account flow?

Reply:
@SammyO That shared preference file is for SINGLE account mode (you won't need it if you operate in Multiple mode)

Reply:
@shahzaibj I think this relates to a conversation we had before: native auth will be in single mode only for now, but there is no use of broker. Is this needed then? Practically in our code currently we don't use it, so I'm inclined to remove it, but I'm not sure if I'm overlooking something.